E2E-ORS-Sample-For-Java/src/main/java/objectReplicationMonitor.java

```diff
@@ -0,0 +1,297 @@
+        /*
+        This method is to archive the replicated blobs with batch
+         */
+
+        System.out.println("\nTransferring replicated blobs in destination container to archive tier...");
+        System.out.println("This may take a few minutes...");
+
+        // Get blobServiceClient to create batch client
+        BlobServiceClient blobServiceClient = new BlobServiceClientBuilder().connectionString(destinationConnectionString).buildClient();
+
+        // Create a batch client that can be used to archive multiple blobs
+        BlobBatchClient blobBatchClient = new BlobBatchClientBuilder(blobServiceClient).buildClient();
+
+        // Get all blobs in container's urls and add them to a list
+        List<String> blobUrls = new ArrayList<>();
+        for (Object b : blobList) {
+            String url = blobServiceClient.getBlobContainerClient(destinationContainer).getBlobClient((String) b).getBlobUrl();
+            blobUrls.add(url);
+        }
+
+        // Set the access tier of the batch to Archive with try/catch
+        try {
+            blobBatchClient.setBlobsAccessTier(blobUrls, AccessTier.ARCHIVE).forEach(response ->
+                    System.out.printf("Setting blob access tier with URL %s completed with status code %d%n",
+                            response.getRequest().getUrl(), response.getStatusCode()));
+
+            System.out.println("\nArchived replicated blobs!");
+        }
+        catch (BlobBatchStorageException ex) {
+            System.out.println("\nThis error may have to do with TLS. Please ensure that you are using 'https' in your connection string rather than 'https', and that 'Secure Transfer' is disabled in your storage account. See README.md for more help.");
+            for (BlobStorageException e : ex.getBatchExceptions()) {
+                String message = e.getServiceMessage();
+                System.out.println(message);
+            }
+        }
+    }
+
+
+    private static void archiveReplicatedDataIndividually(String destinationConnectionString, String destinationContainer, List<String> blobList) {
+        /*
+        This is to archive the data without batch
+         */
+
+        System.out.println("\nTransferring replicated blobs in destination container to archive tier...");
+        System.out.println("This may take a few minutes...");
+
+        // Creating blobserviceclient in order to make containerclient
+        BlobServiceClient sourceBlobServiceClient = new BlobServiceClientBuilder().connectionString(destinationConnectionString).buildClient();
+
+        // Get the container client
+        BlobContainerClient sourceContainerClient = sourceBlobServiceClient.getBlobContainerClient(destinationContainer);
+        if (!sourceContainerClient.exists()) {
+            sourceContainerClient.create();
+        }
+
+        // Iterate through list of blobs
+        for (Object b : blobList) {
```

Comment:
I think now you should be able to use String b: and on line 293 you can remove the (String)

Reply:
Yep! You are right! I still had this same thing earlier in the program too, so I'm glad you caught this one